main.go

```diff
@@ -4,37 +4,87 @@ import (
 	"context"
 	"os"
 
+	producer "github.com/asecurityteam/component-producer"
+	"github.com/asecurityteam/nexpose-vuln-filter/pkg/domain"
 	"github.com/asecurityteam/nexpose-vuln-filter/pkg/filter"
 	v1 "github.com/asecurityteam/nexpose-vuln-filter/pkg/handlers/v1"
-	"github.com/asecurityteam/runhttp"
 	"github.com/asecurityteam/serverfull"
 	"github.com/asecurityteam/settings"
 )
 
-func main() {
-	source, err := settings.NewEnvSource(os.Environ())
+type config struct {
+	Producer            *producer.Config
+	VulnerabilityFilter *filter.VulnerabilityFilterConfig
+	LambdaMode          bool `description:"Use the Lambda SDK to start the system."`
+}
+
+func (*config) Name() string {
+	return "nexposevulnfilter"
+}
+
+type component struct {
+	Producer            *producer.Component
+	VulnerabilityFilter *filter.VulnerabilityFilterComponent
+}
+
+func newComponent() *component {
+	return &component{
+		Producer:            producer.NewComponent(),
+		VulnerabilityFilter: filter.NewVulnerabilityFilterComponent(),
+	}
+}
+
+func (c *component) Settings() *config {
+	return &config{
+		Producer:            c.Producer.Settings(),
+		VulnerabilityFilter: c.VulnerabilityFilter.Settings(),
+	}
+}
+
+func (c *component) New(ctx context.Context, conf *config) (func(context.Context, settings.Source) error, error) {
+	f, err := c.VulnerabilityFilter.New(ctx, conf.VulnerabilityFilter)
```

Comment:
Suggestion: This is probably outside the scope of this PR, but I wanted to call out that it appears that the `VulnerabilityFilterComponent.New()` func misleadingly returns a `VulnerabilityFilterCriteria` struct. The actual `VulnerabilityFilter` appears to be part of the handler, and is very tightly coupled with the `VulnerabilityFilterCriteria`. I'd suggest
1.  Renaming the `VulnerabilityFilterComponent` to `VulnerabilityFilterCriteriaComponent` in pkg/filter/resource.go
2. Renaming the field in the component/config structs in this file
3. Opening a separate issue to move the VulnerabilityFilter out of the handler and into the filter package, and making the filter an interface that is passed to the handler which encapsulates all of the logic for filtering (which would include any criteria).

Reply:
Right, good call out there. I thought about doing that very thing, but didn't want to bog down this PR. I will create a ticket for that work before moving on. 👍

Reply:
Needed to up the test coverage, so went ahead and cleaned up this part of the tech debt while I was at it.